js/customize-snapshots-admin.js

```diff
@@ -0,0 +1,73 @@
+( function( $ ) {
+	'use strict';
+
+	$( function() {
+		var $link, linkText, linkActions, dataSlug, initializeLink;
+
+		$link = $( '.snapshot-toggle-setting-removal' );
+		linkText = [ 'Remove setting', 'Restore setting' ];
+		linkActions = [ 'remove', 'restore' ];
+		dataSlug = 'cs-action';
+
+		initializeLink = function() {
+			$link.text( linkText[ 0 ] )
+				.data( dataSlug, linkActions[ 0 ] );
+		};
+
+		initializeLink();
+
+		$link.on( 'click', function( event ) {
+			var $clickedLink, $settingDisplay, clickedLinkAction, settingId;
+
+			event.preventDefault();
+
+			$clickedLink = $( this );
+			$settingDisplay = $( this ).parents( 'details' );
+			clickedLinkAction = $( this ).data( dataSlug );
+			settingId = $( this ).attr( 'id' );
+
+			this.isLinkSetToRemoveSetting = function() {
+				return linkActions[ 0 ] === clickedLinkAction;
+			};
+
+			this.hideSettingAndChangeLinkText = function() {
+				$clickedLink.text( linkText[ 1 ] )
+					.data( dataSlug, linkActions[ 1 ] )
+					.after( this.constructHiddenInputWithValue() );
+				$settingDisplay.removeAttr( 'open' )
+					.addClass( 'snapshot-setting-removed' );
+			};
+
+			this.constructHiddenInputWithValue = function() {
```

Comment:
Better for this function to be more of a pure function that doesn't rely on its closure scope. So it can be at the top level as `component.constructHiddenInputWithValue()` and it can take a `settingId` as its argument. It is much easier to test the function like this, and there is only ever one copy, whereas here there is a copy for each time the user clicks.